test/extended/cli/builds.go

```diff
@@ -0,0 +1,339 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	t "github.com/onsi/gomega/types"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	"regexp"
+	"strings"
+	"time"
+)
+
+const (
+	bcOutputTemplate       = "{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}"
+	bcOutputToNameTemplate = "{{ .spec.output.to.name }}"
+	bcSourceTypeTemplate   = "{{ .spec.source.type }}"
+	testDockerFileContent  = "FROM image-registry.openshift-image-registry.svc:5000/openshift/tests:latest"
+)
+
+func waitForBuild(oc *exutil.CLI, bcName, buildNumber string) error {
+	args := []string{"builds"}
+
+	if buildNumber != "" {
+		args = append(args, fmt.Sprintf("%v-%v", bcName, buildNumber))
+	} else {
+		args = append(args, "-l", fmt.Sprintf("openshift.io/build-config.name=%v", bcName))
+	}
+
+	return wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
```

Comment:
First, you don't need to use `PollImmediate` since you're starting a build so I doubt it'll finish that soon, `wait.Poll` first waits and then performs poll which makes more sense in this case.

> I am not sure if I should wait here with `oc` or directly with go API

I'd probably go with an oc call, since we're focusing on that. The other tests are testing client-go calls.